### PR TITLE
8295804: javax/swing/JFileChooser/JFileChooserSetLocationTest.java failed with "setLocation() is not working properly"

### PR DESCRIPTION
**Pull request by TejeshR13**
The issue is that the doc area (in respect to the screen height which is 768px) which is at the bottom was causing the `JFileChooser `to be placed slightly above the set location. Was able to reproduce in local machine with reference to the failure image provided in the CI logs. The suggested fix is to place the main Frame slightly above the center of the screen than setting at the center of the screen. Several CI runs were made and no issue found.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295804](https://bugs.openjdk.org/browse/JDK-8295804): javax/swing/JFileChooser/JFileChooserSetLocationTest.java failed with "setLocation() is not working properly" (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17364/head:pull/17364` \
`$ git checkout pull/17364`

Update a local copy of the PR: \
`$ git checkout pull/17364` \
`$ git pull https://git.openjdk.org/jdk.git pull/17364/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17364`

View PR using the GUI difftool: \
`$ git pr show -t 17364`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17364.diff">https://git.openjdk.org/jdk/pull/17364.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17364#issuecomment-1886242078)

### CODE REVIEW

**Comment by kumarabhi006** [comment]
> The issue is that the doc area (`in respect to the screen width which is 768px`) which is at the bottom was causing the JFileChooser to be placed slightly above the set location.

Is it with respect to height ?

**Comment by TejeshR13** [comment]
> > The issue is that the doc area (`in respect to the screen width which is 768px`) which is at the bottom was causing the JFileChooser to be placed slightly above the set location.
> 
> Is it with respect to height ?

Yeah, corrected now.

**Comment by kumarabhi006** [comment]
Does the test fails only on the system which is 768px in height or in any other screen size also?

**Comment by TejeshR13** [comment]
> Does the test fails only on the system which is 768px in height or in any other screen size also?

Not only on 768, whichever size isn't enough for the FileChooser to show up fully it'll fail. I was able to check till 720px in local machine where it fails. After fix I tested in all C machines and it is green.

File: test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java [hunk 1]
```diff
@@ -209,6 +210,10 @@ private static void hitKeys(int... keys) {
     }
 
     public static void createUI() {
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+
+        int screenWidth = (int) screenSize.getWidth() / 2;
+        int screenHeight = (int) screenSize.getHeight() / 2;
         frame = new JFrame();
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java:217]: Frame should have title.

File: test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java [hunk 2]
```diff
@@ -209,7 +210,11 @@ private static void hitKeys(int... keys) {
     }
 
     public static void createUI() {
-        frame = new JFrame();
+        Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+
+        int width = (int) screenSize.getWidth() / 2;
+        int height = (int) screenSize.getHeight() / 2;
```
Comment (mrserb) [inline test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java:216]: This is not the width and height, this is the kind of point(x,y) where the frame will be located.

Comment (TejeshR13) [reply test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java]: Yeah, I got it now. Updated.

**Review by prrace — APPROVED** [review]
I suppose you can't know exactly how much space you need since the dialog size is out of your control.

**Review by mrserb — APPROVED** [review]

**Comment by TejeshR13** [comment]
> I suppose you can't know exactly how much space you need since the dialog size is out of your control.

Yes, I've fixed screen center position - 200 based on my test runs with multiple screen size.

**Pull request closed without merging**